torch/distributed/fsdp/fully_sharded_data_parallel.py

```diff
@@ -1139,6 +1138,7 @@ def _register_param_handles_from_handles(
         self._deregister_flat_params()
         self._handles.clear()
         self.params.clear()
+        self._module_to_handles = collections.defaultdict(list)
```

Comment:
we need to reconstruct `_module_to_handles` here

Reply:
removed this line, since we now clear `_module_to_handles` in `_construct_module_to_handles`